internal/fields/mappings.go

```diff
@@ -542,27 +544,28 @@ func (v *MappingValidator) compareMappings(path string, preview, actual map[stri
 		if err != nil {
 			errs = append(errs, fmt.Errorf("found invalid multi_fields type in actual mappings for path %q: %w", path, err))
 		}
-		compareErrors := v.compareMappings(path, previewFields, actualFields)
+		compareErrors := v.compareMappings(path, false, previewFields, actualFields)
 		errs = append(errs, compareErrors...)
 		// not returning here to keep validating the other fields of this object if any
 	}
 
 	// Compare and validate the elements under "properties": objects or fields and its parameters
-	propertiesErrs := v.validateObjectProperties(path, containsMultifield, actual, preview)
+	propertiesErrs := v.validateObjectProperties(path, true, containsMultifield, preview, actual)
 	errs = append(errs, propertiesErrs...)
 	if len(errs) == 0 {
 		return nil
 	}
 	return errs.Unique()
 }
 
-func (v *MappingValidator) validateObjectProperties(path string, containsMultifield bool, actual, preview map[string]any) multierror.Error {
+func (v *MappingValidator) validateObjectProperties(path string, couldBeParametersDefinition, containsMultifield bool, preview, actual map[string]any) multierror.Error {
```

Comment:
Reordered preview and actual parameters to keep consistency with the other methods.